dessia_common/datatools/dataset.py

```diff
@@ -210,9 +210,9 @@ def _printed_attributes(self):
 
     def _print_objects_slice(self, key: slice, attr_space: int):
         string = ""
-        for dessia_object in self.dessia_objects[key]:
+        for dessia_object_idx in range(len(self[key])):
```

Comment:
I think just `i` would be justified here as a loop variable, since we loop in a range. It would be less cumbersome, in my opinion.

As a sidenote, indexing self bugs me out a bit. I don't find any reason why it would be bad, as long as self behaves as a sequence, but I am a bit puzzled by it and find it weird. Do not change anything for now, because I think this is OK, but we might consider explicitly using `__getitem__` in the future

Reply:
I guess pylint is not happy when using __getitem__ instead of self indexing, because it is not happy when you write self.__len__(), saying that you should write len(self).

Reply:
Ok, for these kind of cases we can let pylint complain or not watch the error, but we will see this later